common/src/main/java/com/graphicsfuzz/common/util/PipelineInfo.java

```diff
@@ -310,6 +310,15 @@ public void addComputeInfo(int numGroupsX, int numGroupsY, int numGroupsZ,
     dictionary.add(Constants.COMPUTE_DATA_KEY, computeData);
   }
 
+  public void addGridInfo(int columns, int rows) {
+    final JsonObject gridDimensions = new JsonObject();
```

Comment:
If you add `hasGridInfo()` as per my suggestion below, you could assert `!hasGridInfo()` here.

Reply:
Added assert.